istioctl/cmd/describe.go

```diff
@@ -427,6 +428,14 @@ func printPod(writer io.Writer, pod *v1.Pod) {
 			}
 			ports = append(ports, fmt.Sprintf("%d%s (%s)", port.ContainerPort, protocol, container.Name))
 		}
+		// Ref: https://istio.io/latest/docs/ops/deployment/requirements/#pod-requirements
```

Comment:
This puts the onus on the user to not run as UID 1337. We have a fix in Maistra that prevents this from becoming a problem for the user by just using a different UID for proxy than the user uses for their pod: https://github.com/maistra/istio/pull/230/files. As a bonus, I think it's more secure too. Should we consider that?

Reply:
I would rather see a policy so that it is impossible for the app container to use 1337.  If the agent's UID bounces around, it is tricky to debug.

Reply:
Neither this check nor the one in Maistra actually work in all cases. A user can set the `USER` command in the Dockerfile.

Aside from that, I am against the change to dynamically provision. Aside from debugging issues, its a breaking change that will break many many users.